### PR TITLE
If hashing more than 2^53-1 bits, throw exception, #299

### PR DESCRIPTION
**Pull request by ph4r05**

### CODE REVIEW

File: core/sha1.js [hunk 1]
```diff
@@ -36,6 +36,12 @@ sjcl.hash.sha1.prototype = {
    * @constant
    */
   blockSize: 512,
+
+  /**
+   * Maximal number of bits this function can hash.
+   * @constant
+   */
+  maxSafeLength: Math.pow(2, 53) - 1, //Number.MAX_SAFE_INTEGER
```
Comment (Nilos) [inline core/sha1.js]: why not just use the number here?
`9007199254740991`
Calculating this every time sjcl loads seems a lot of work.

Comment (ph4r05) [inline core/sha1.js]: OK I will re-do it

Comment (Nilos) [inline core/sha1.js]: Thanks!

**Comment by Nilos** [comment]
👍  thanks will merge as soon as Travis is green.

**Comment by ph4r05** [comment]
I can squash to commit to one, if you want. to make commit history cleaner.

**Comment by ph4r05** [comment]
I've squashed it to one commit

**Pull request merged**